rfc/rfc-44/rfc-44.md

```diff
@@ -0,0 +1,156 @@
+to retrieve Hudi table metadata. Besides, HudiMetadata takes use of HoodieTableMetaClient to expose Hudi specified
+metadata, e.g. instants list of a table timeline, represented as SystemTable.
+      
+### HudiSplitManager & HudiSplit
+
+HudiSplitManager implements the ConnectorSplitManager interface. It partitions a Hudi table to read into multiple
+individual chunks (called ConnectorSplit in Presto), so that the data set can be processed in parallel.
+HudiSplitManager also performs partition  pruning if possible.
+
+HudiSplit, which implements ConnectorSplit, describes which files to read, and the file properties (i.e. offset, length,
+location, etc).  For a normal Hudi table, HudiSplitManager generates a split for each FileSlice at certain instant.
+For a Hudi table that contains large base files, HudiSplitManager divides the base file into multiple splits,
+each containing all the log files and part of the base file. This should improve the performance.
+
+For large Hudi tables, generating splits tends to cost lots of time. This can be improved in the following ways. 
+Firstly, increase parallelism. Since each partition is independent of another, processing the partitions in multiple
+threads, one thread per partition, helps reduce the wall time. Secondly, take use of Hudi metadata table or external
+listing services. This should save a large amount of cost to make requests to HDFS.
+
+
+### HudiPageSourceProvider & HudiPageSource
+
+For each HudiSplit, HudiPageSourceProvider generates a reader for it to load data from the data source (e.g. HCFS)
+into memory. Presto organizes the memory data in the form of Pages. Page is implemented in a column-store favor. 
+This design contributes to enhance Presto's query performance. In particular, Presto writes its own readers to
+accelerate the data loading for column-store favored files, such as Parquet and ORC.
+
+For the snapshot query on a COW table, HudiPageSourceProvider takes advantage of Presto's native ParquetPageSource
+as it's well optimized for Parquet files as described above.
+
+For the snapshot query on an MOR table, HudiPageSourceProvider is planned to add a page source that makes mixed uses of
+ParquetPageSource and HoodieRealtimeRecordReader. The new page source takes care of memory usage under Presto's memory
+tracking framework. A native log file reader optimized for Presto might be provided in the future.
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
+  
+  There will be no impact on existing users because this is a new connector. It does not change the behavior of current
+  integration through the existing Hive connector. It gives users more choice.
+
+- What do we lose if we move away from the Hive connector?
+
+  Some features in Hive connectors (e.g. RaptorX) might be temporarily unavailable during the early development stage of
```

Comment:
Apart from caching, are going to miss out something on filter evaluation? Will the filter evaluation be done by the presto engine instead of hive connector?

Reply:
What is the exact meaing here for `filter evaluation`? 

AFAI, presto push down some predicate expresssions to the connectors. Connectors take use of the expressions to prune some chunks of input data. For hudi connector, we could do it as well.